manager/backend/api.py

```diff
@@ -1,65 +1,175 @@
+from django.db.models import query
 from .models import Author, Post
 from rest_framework import serializers, viewsets, permissions, generics
 from rest_framework.response import Response
 from .serializers import AuthorSerializer, RegisterSerializer, UserSerializer, PostSerializer
 from rest_framework.authtoken.models import Token
 
-# Get Author API
+class RegisterAPI(generics.GenericAPIView):
+	"""
+	This class provides an API for user and author registration.
+	"""
+
+	serializer_class = RegisterSerializer
+
+	def post(self, request, *args, **kwargs):
+		"""
+		This method provides POST functionality to creating a user and author, by serializing a user object that is created into the Django authentication system, then creating a POST request with the author data.
+		"""
+		# Get and serialize the request data
+		serializer = self.get_serializer(data=request.data)
+
+		# Check the validity
+		serializer.is_valid(raise_exception=True)
+
+		# Create the user
+		user = serializer.save()
+		# Create an authentication token for the provided user
+		token = Token.objects.create(user=user)
+
+		# Create the author object
+		author = Author(
+						token=token,
+						displayName=request.data["displayName"],
+						github=request.data["github"],
+						host = request.META['HTTP_HOST'],
+						)
+
+		# Save the author information into the database
+		author.url = "http://"+str(author.host)+"/"+str(author.id)
+		author.save()
+		# Serialize the author data for a POST response
+		authorData = AuthorSerializer(author, context=self.get_serializer_context()).data
+
+		return Response(authorData)
+
+class UserAPI(generics.RetrieveAPIView):
+
+	permission_classes = [
+		permissions.IsAuthenticated,
+	]
+
+	serializer_class = UserSerializer
+
+	def get_object(self):
+		return self.request.user
+
+
+
+
 class AuthorViewSet(viewsets.ModelViewSet):
-    
-    queryset = Author.objects.all()
+	"""
```

Comment:
when we edit an author's info it will generate a new author id for him/her?

Reply:
I wrote this earlier as well, but in it's current state this will not create a new author id. It will only allow edits if the correct user id is present in the request. I could change it to work like that most likely but is that the way we want to move forward? I tried looking in the spec but can't find any info, let me know if you have any more info.

Reply:
No we do not want to change the author id ever I think.

Reply:
Just made sure it's not on the Author model, thanks for pointing that out!